apps/omg_eth/test/omg_eth/root_chain_test.exs

```diff
@@ -0,0 +1,155 @@
+
+    hex_to_binary = fn (hex) ->
+      {:ok, result} = hex |> String.replace_prefix("0x", "") |> Base.decode16(case: :lower)
+      result
+    end
+
+    contract = %{
+      # NOTE: achiurizo
+      # this has changed with ALD(see plasma-contrats deploy of plasma_framework)
+      # it's now :plasma_framework_tx_hash instead of :txhash_contract
+      txhash_contract: hex_to_binary.("0x3d517d431daea71a99310f12468ffdf2bf547ad1d148f42acfc4ee34dd4e84d7"),
+      plasma_framework: hex_to_binary.("0xd17e1233a03affb9092d5109179b43d6a8828607"),
+      eth_vault: hex_to_binary.("0x1967d06b1faba91eaadb1be33b277447ea24fa0e"),
+      erc20_vault: hex_to_binary.("0xaef6182310e3d34b6ea138b60d36a245386f3201"),
+      payment_exit_game: hex_to_binary.("0x902719f192aa5240632f704aa7a94bab61b86550"),
+      authority_address: hex_to_binary.("0x22d491bde2303f2f43325b2108d26f1eaba1e32b")
+    }
+    {:ok, contract: contract}
+  end
+
+  test "get_root_deployment_height/2 returns current block number", %{contract: contract} do
+    use_cassette "ganache/get_root_deployment_height", match_requests_on: [:request_body] do
+      {:ok, number} = RootChain.get_root_deployment_height(contract.txhash_contract, contract)
+      assert is_integer(number)
+    end
+  end
+
+  test "get_next_child_block/1 returns next blknum to be mined by operator", %{contract: contract} do
+    use_cassette "ganache/get_next_child_block", match_requests_on: [:request_body] do
+      assert {:ok, 1000} = RootChain.get_next_child_block(contract)
+    end
+  end
+
+  describe "has_token/2" do
+    # TODO achiurizo
+    #
+    # Figure out why I can't use the same cassettes even though request_body is unique
+    @tag :skip
+    test "returns true  if token exists", %{contract: contract} do
+      use_cassette "ganache/has_token_true", match_requests_on: [:request_body] do
+        assert {:ok, true} = RootChainHelper.has_token(@eth, contract)
+      end
+    end
+
+    # TODO achiurizo
+    #
+    # Skipping these specs for now as this function needs to be updated
+    # to use the new ALD function (not hasToken?)
+    @tag :skip
+    test "returns false if no token exists", %{contract: contract} do
+      use_cassette "ganache/has_token_false", match_requests_on: [:request_body] do
+        assert {:ok, false} = RootChainHelper.has_token(<<1::160>>, contract)
+      end
+    end
+  end
+
+  test "get_child_chain/2 returns the current block hash and timestamp", %{contract: contract} do
+    use_cassette "ganache/get_child_chain", match_requests_on: [:request_body] do
+      {:ok, {child_chain_hash, child_chain_time}} = RootChain.get_child_chain(0, contract)
+
+      assert is_binary(child_chain_hash)
+      assert byte_size(child_chain_hash) == 32
+      assert is_integer(child_chain_time)
+    end
+  end
+
+  test "submit_block/1 submits a block to the contract", %{contract: contract} do
+    use_cassette "ganache/submit_block", match_requests_on: [:request_body] do
+      block =
```

Comment:
this doesn't return a block, but `{:ok, txhash}` - a response from the `send_transaction` call underneath. It is actually intended to be piped into the `transact_sync!`, so you don't have to think about the correct name for this transient thing :).

Reply:
Transact sync has a tuple argument so that it's able to be piped, but on the other hand, if it's meant to be piped it doesn't make sense for `submit_block` to return a tuple. So that's a little bit odd.

So I would say let's keep the behavior that @achiurizo introducted and break `transacy_sync!` so that it's consistent why how pipes should pass and transform arguments.

`{:ok, transaction_hash} = RootChain.submit_block(...`
`assert {:ok, _} = DevHelpers.transact_sync!(transaction_hash)`

and a fix to `dev_helpers`:
```
@doc """
  Use with contract-transacting functions that return {:ok, txhash}, e.g. `Eth.Token.mint`, for synchronous waiting
  for mining of a successful result
  """
  @spec transact_sync!(Eth.hash()) :: {:ok, map}
  def transact_sync!(transaction_hash) do
    {:ok, %{"status" => "0x1"} = result} = WaitFor.eth_receipt(transaction_hash, @about_4_blocks_time)
    {:ok, Map.update!(result, "blockNumber", &int_from_hex(&1))}
  end
```

Reply:
regardless of where this thread takes us - the `block` name is still confusing. Pls rename to `submit_block_response`